cardano-base/src/Cardano/Base/FeatureFlags.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DerivingStrategies #-}
+
+-- | Mini-abstraction to aid parallel development of experimental Cardano
+-- features.
+--
+-- == Usage example
+--
+-- >>> import Data.Set (Set)
+-- >>> import qualified Data.Set as Set
+-- >>> import Cardano.Base.FeatureFlags (CardanoFeatureFlag (..))
+--
+-- >>> :{
+-- logic :: Set CardanoFeatureFlag -> Int -> Int
+-- logic featureFlags
+--   | Set.member PerasFlag featureFlags = (+ 2)
+--   | otherwise = (+ 1)
+-- :}
+module Cardano.Base.FeatureFlags (
+  CardanoFeatureFlag (..),
+) where
+
+import GHC.Generics (Generic)
+import NoThunks.Class (NoThunks)
+
+-- | An experimental Cardano feature.
+data CardanoFeatureFlag
+  = -- | Feature flag for Ouroboros Peras (faster settlement).
+    PerasFlag
+  | -- | Feature flag for Ouroboros Leios (higher throughput).
+    LeiosFlag
+  | -- | Feature flag for Ouroboros Phalanx (anti-grinding).
+    PhalanxFlag
+  deriving stock (Show, Read, Eq, Ord, Generic)
+  deriving anyclass (NoThunks)
```

Comment:
Any further type classes to derive? Maybe `{From,To}JSON` (brings in an `aeson` dependency)?

Reply:
Yeah, `aeson` is a fine dependency to add.

Reply:
> Enum/Bounded?

Yes, please!